File: benchmark/lua/lcs.lua
local start = os.clock()
local s1 = string.rep("abcdefghij", 10)
local s2 = string.rep("acegikmoqs", 10)
local m, n = #s1, #s2
local dp = {}
for i = 0, m do
    dp[i] = {}
    for j = 0, n do dp[i][j] = 0 end
end
for i = 1, m do
    for j = 1, n do
        if s1:sub(i, i) == s2:sub(j, j) then
            dp[i][j] = dp[i - 1][j - 1] + 1
        else
            local a = dp[i - 1][j]
            local b = dp[i][j - 1]
            dp[i][j] = a > b and a or b
        end
    end
end
local finish = os.clock()
print("LCS: " .. dp[m][n])
print(string.format("Time: %.0fms", (finish - start) * 1000))

File: benchmark/perl/lcs.pl
use Time::HiRes qw(time);
my $s1 = "abcdefghij" x 10;
my $s2 = "acegikmoqs" x 10;
my $start = time();
my ($m, $n) = (length($s1), length($s2));
my @dp;
for my $i (0..$m) { for my $j (0..$n) { $dp[$i][$j] = 0; } }
for my $i (1..$m) {
    for my $j (1..$n) {
        if (substr($s1,$i-1,1) eq substr($s2,$j-1,1)) {
            $dp[$i][$j] = $dp[$i-1][$j-1] + 1;
        } else {
            $dp[$i][$j] = $dp[$i-1][$j] > $dp[$i][$j-1] ? $dp[$i-1][$j] : $dp[$i][$j-1];
        }
    }
}
my $end = time();
my $ms = ($end - $start) * 1000;
print "LCS: $dp[$m][$n]
";
printf "Time: %.0fms
", $ms;
